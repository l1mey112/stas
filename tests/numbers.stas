include <std.stas>

fn main {
	-1                   putuln
	-100                 putuln
	-9223372036854775808 putuln
	18446744073709551615 putuln
	100 -50 -            putuln

	-10   0  <s if { 'less\n'           puts } else { 'not\n' puts }
	-10   0  >s if { 'greater\n'        puts } else { 'not\n' puts }
	 10   0  <s if { 'less\n'           puts } else { 'not\n' puts }
	 10   0  >s if { 'greater\n'        puts } else { 'not\n' puts }
	-10 -10 <=s if { 'less equals\n'    puts } else { 'not\n' puts }
	-10 -10 >=s if { 'greater equals\n' puts } else { 'not\n' puts }

	5992 putuln
	5994 putuln
}