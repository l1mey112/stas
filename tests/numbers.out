18446744073709551615
18446744073709551516
9223372036854775808
18446744073709551615
150
less
not
not
greater
less equals
greater equals
5992
5994
